### ISSUE TITLE
Can't access app settings when URL is not reachable

### ISSUE THREAD

**Issue by bartkummel** [home-assistant/iOS#1929]
**iOS device model, version and app version**

Model Name: iPhone Xs
Software Version: iOS 15.0.2
App version: 2021.11.1 (2021.266)

**Home Assistant Core Version**

2021.10.7

**Describe the bug**
When the (local or remote) URL is not reachable for whatever reason, I can't access the **app settings**, because there is not menu shown. This is a problem, because I want to change the URL.

**To Reproduce**
Configure a non-existing URL als the Local URL. Restart the app. Now you can't change the URL back to a valid one.

**Expected behavior**
There should be a link to the **app settings** screen on the "no connection" screen.

**Screenshots**

**Additional context**

As a work around, if you have configured an external URL that is still valid, you can turn off the WiFi and connect via the cellular network. Then the menu is available and you can edit the internal URL again.

**Comment by zacwest** [comment]
Tap OK on the error banner and the settings cog will be available.

**Comment by bartkummel** [comment]
You're right, I should have tested it mor thoroughly. The problem arises when the URL _is_ valid, but there are connection problems, like a timeout. Especially when that happens when the app is restarted, you are stuck in a modal error message, no cog button or menu available.

**Comment by zacwest** [comment]
Can you share a screenshot of this problem? I'm not sure I understand.

**Comment by bartkummel** [comment]
I tried to reproduce it, but now I get the red error bar with the OK button on it. I'll post a screenshot if it occurs again. It was an all-white page, with an error message in a small black font, vertically and horizontally centered on the page. There were not buttons or links, so no interaction was possible.

**Comment by zacwest** [comment]
Hm, this may be after the app has successfully connection with the frontend but before the frontend finishes loading. It may be worth trying to use a different point in the frontend lifecycle to start hiding it. You can likely pull-to-refresh to get rid of it.

Medium-term, I'm planning on killing the settings icon in the bottom-right in favor of an always-visible element at the top, so the problem may also sort itself out.

**Comment by bartkummel** [comment]
Pull-to-refresh didn't resolve it, I tried that.

I think an always-visible element would be a great solution.
